File: src/com/gb/javalearn/lesson9final/Lesson9task68.java
package com.gb.javalearn.lesson9final;

import java.util.Scanner;

/*
 Задача 68: Напишите программу вычисления функции Аккермана с помощью рекурсии. Даны два неотрицательных числа m и n.
 Task 68: Write a program to calculate the Ackermann function using recursion. Two non-negative numbers m and n are given.
 */
public class Lesson9task68 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Введите число m: ");
        int m = scanner.nextInt();
        System.out.print("Введите число n: ");
        int n = scanner.nextInt();
        int result = ackermann(m, n);
        System.out.printf("A(%d,%d) = %d\n", m, n, result);
    }

    public static int ackermann(int m, int n) {
        if (m == 0) {
            return n + 1;
        } else if (n == 0) {
            return ackermann(m - 1, 1);
        } else {
            return ackermann(m - 1, ackermann(m, n - 1));
        }
    }
}

File: src/com/gb/javalearn/lesson9final/CSharp/Lesson9_task68.cs
using System;

class Lesson9_task66 {
  public static void Main (string[] args) {
    Console.Write("Введите число m: ");
    int m = int.Parse(Console.ReadLine());
    Console.Write("Введите число n: ");
    int n = int.Parse(Console.ReadLine());
    int result = Ackermann(m, n);
    Console.WriteLine($"A({m},{n}) = {result}");
  }

  public static int Ackermann(int m, int n) {
    int result = 0;
    if (m == 0)
      result = n + 1;
    else if (n == 0)
      result = Ackermann(m - 1, 1);
    else
      result = Ackermann(m - 1, Ackermann(m, n - 1));

    return result;
  }
}
